scripts/installer.sh

```diff
@@ -0,0 +1,251 @@
+			;;
+ 		arch)
+			echo "Please install Tailscale from the community repo:"
+			echo
+			echo "pacman -S tailscale"
+			echo "systemctl enable --now tailscaled"
+			exit 2
+			;;
+		manjaro)
+			echo "Please install Tailscale from the community repo:"
+			echo
+			echo "pacman -S tailscale"
+			echo "systemctl enable --now tailscaled"
+			exit 2
+			;;
+		alpine)
+			echo "Please install Tailscale from the alpine testing repo"
+			echo
+			echo "apk add tailscale"
+			echo "rc-update add tailscale"
+			exit 2
+			;;
+		nixos)
+			echo "Please add Tailscale to your NixOS configuration directly"
+			echo
+			echo "services.tailscale.enable = true;"
+			exit 2
+			;;
+		void)
+			echo "Please install Tailscale from the void current repo"
+			echo
+			echo "xbps-install tailscale"
+			exit 2
+			;;
+		gentoo)
+			echo "Please install Tailscale from portage"
+			echo
+			echo "emerge net-vpn/tailscale"
+			exit 2
+			;;
+		freebsd)
+			echo "Please install Tailscale using pkg"
+			echo
+			echo "pkg add tailscale"
+			exit 2
+			;;
+		openbsd)
+			echo "Tailscale does not currently have packages for OpenBSD."
+			echo "Please install Tailscale from source: https://github.com/tailscale/tailscale"
+			exit 2
+			;;
+		*)
+			echo "Your OS ($ID) is not supported by this script."
+			echo "Please contact us at support@tailscale.com for help."
+			exit 2
+			;;
+		# TODO: wsl?
+	esac
+
+	# Step 2: having detected an OS we support, is it one of the
+	# versions we support?
+	OS_UNSUPPORTED=
+	case "$OS" in
+		ubuntu)
+			if [ "$VERSION" != "16.04" -a
+				 "$VERSION" != "18.04" -a
+				 "$VERSION" != "19.10" -a
+				 "$VERSION" != "20.04" -a
+				 "$VERSION" != "20.10" -a
+				 "$VERSION" != "21.04"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		debian)
+			if [ "$VERSION" != "stretch" -a
+				 "$VERSION" != "buster" -a
+				 "$VERSION" != "bullseye" -a
+				 "$VERSION" != "sid"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		raspbian)
+			if [ "$VERSION" != "buster"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		centos)
+			if [ "$VERSION" != "7" -a
+				 "$VERSION" != "8"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		rhel)
+			if [ "$VERSION" != "8"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		amazon-linux)
+			if [ "$VERSION" != "2"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		opensuse)
+			if [ "$VERSION" != "leap/15.1" -a
+				 "$VERSION" != "leap/15.2" -a
+				 "$VERSION" != "tumbleweed"
+			   ]; then
+				OS_UNSUPPORTED=1
+			fi
+		;;
+		*)
+			OS_UNSUPPORTED=1
+	esac
+	if [ "$OS_UNSUPPORTED" = "1" ]; then
+		echo "$OS $VERSION is not a supported version of $OS for Tailscale."
+		echo "Please contact us at support@tailscale.com for help."
+		exit 2
+	fi
+
+	echo "Installing Tailscale for $OS $VERSION, using $PACKAGETYPE"
+
+	case "$PACKAGETYPE" in
+		apt)
+			# TODO: use newfangled per-repo signature scheme
+			set -x
+			curl -fsSL "https://pkgs.tailscale.com/stable/$OS/$VERSION.gpg" | sudo apt-key add -
```

Comment:
No time like the present, I suppose: Debian Bullseye and Ubuntu 21.x want us to not use apt-key any more.

Details and history in https://github.com/tailscale/tailscale/issues/1876

Reply:
I'll do it as a followup, this is a more or less straight port of our existing instructions.